kani-driver/src/cbmc_property_renderer.rs

```diff
@@ -421,6 +421,86 @@ pub fn format_result(
     result_str
 }
 
+/// Formats a coverage result item (i.e., the subset of verification checks with coverage property class).
+/// To be used when the user requests coverage information with --coverage. The output is tested through the coverage-based testing suite, not the regular expected suite.
+/// Loops through each of the check with a coverage property class and gives a status of COVERED if all checks pertaining
+/// to a line number are SATISFIED. Otherwise, it gives a status of UNCOVERED.
+pub fn format_result_coverage(properties: &[Property]) -> String {
+    let mut formatted_output = String::new();
+    formatted_output.push_str("\nCoverage Results:\n");
+
+    let coverage_checks: Vec<Property> =
+        properties.iter().filter(|&x| x.property_class() == "coverage").cloned().collect();
+
+    let mut sorted_checks: Vec<&Property> = coverage_checks.iter().collect();
+
+    sorted_checks.sort_by_key(|check| (&check.source_location.file, &check.source_location.line));
+
+    let mut files: HashMap<String, Vec<(usize, String)>> = HashMap::new();
+    for check in sorted_checks {
+        // Get line number and filename
+        let line_number: usize = check.source_location.line.as_ref().unwrap().parse().unwrap();
+        let file_name: String = check.source_location.file.as_ref().unwrap().to_string();
+
+        // Add to the files lookup map
+        if !files.contains_key(&file_name) {
+            files.insert(file_name.clone(), vec![(line_number, check.status.clone().to_string())]);
+        } else {
+            files
+                .get_mut(&file_name)
+                .unwrap()
+                .push((line_number, check.status.clone().to_string()));
+        }
+    }
+
+    let mut coverage_results: HashMap<String, Vec<(usize, String)>> = HashMap::new();
+
+    // loop through the files list and create a list of lines accessible in that file
+    for file in files.keys() {
+        let mut lines: HashSet<usize> = HashSet::new();
+        let mut line_results: Vec<(usize, String)> = Vec::new();
+        for check in files.get(file).unwrap() {
```

Comment:
You can change the loop header to:
```rust
for (file, val) in files
```
to avoid doing a `get(file)` to get the value.